libraries/chain/controller.cpp

```diff
@@ -930,7 +939,7 @@ struct controller_impl {
         cfg.state_size, false, cfg.db_map_mode ),
     blog( cfg.blocks_dir, cfg.blog ),
     block_data(block_data_t::block_data_variant{
-         std::in_place_type<block_data_legacy_t>, // [greg todo] create correct type depending on whether IF activated
+         std::in_place_type<block_data_legacy_t>, // initial state is always dpos
```

Comment:
Not sure that will be true after IF is enabled on startup, but fine for now.

Reply:
OK. I thought I asked and we said we would always go through the transition, but I was thinking it might not be true when we start from a snapshot.

Reply:
We will always go through the transition when starting a new chain. We have no immediate plans to add the feature of enabling protocol feature activation in genesis. 
Startup of nodeos after transition will start with state (either from chainbase or snapshot) that is in IF mode.